docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,752 @@
+    "qnn.compose(ansatz, inplace=True)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "outputs": [],
+   "source": [
+    "# discriminator = Discriminator()"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 50,
+   "outputs": [],
+   "source": [
+    "# d_params = discriminator."
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 51,
+   "outputs": [],
+   "source": [
+    "# nn.init.zeros_(d_params)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Definition of the loss functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 52,
+   "outputs": [],
+   "source": [
+    "# Loss function\n",
+    "g_loss_fun = nn.BCELoss()\n",
+    "d_loss_fun = nn.BCELoss()"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Evaluation of custom gradients for the generator BCE loss function"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 53,
+   "outputs": [],
+   "source": [
+    "def g_loss_fun_grad(qnn: QuantumCircuit,\n",
+    "                    parameters: Union[ParameterVector, ParameterExpression, List[ParameterExpression]],\n",
+    "                    param_values: Iterable,\n",
+    "                    discriminator_: nn.Module,\n",
+    "                    grad_method: str = 'param_shift') -> Iterable:\n",
+    "    \"\"\"\n",
+    "    Custom gradient of the generator loss function considering the custom gradients of the quantum generator\n",
+    "    Args:\n",
+    "        qnn: Quantum neural network ansatz given as a quantum circuit.\n",
+    "        parameters: The parameters of the quantum neural network which are trained.\n",
+    "        param_values: The current values of the quantum neural network parameters.\n",
+    "        discriminator_: Classical neural network representing the discriminator.\n",
+    "        grad_method: Method used to compute the gradients {'param_shift', 'lin_comb', 'fin_diff'}\n",
+    "    Returns:\n",
+    "        List of gradient values, i.e., the gradients of the loss function w.r.t. the quantum neural network parameters\n",
+    "    \"\"\"\n",
+    "    grads = generator_grad(qnn, parameters, param_values, grad_method = grad_method)\n",
```

Comment:
as stated before, we could just remove `generator_grad` and put the code directly here